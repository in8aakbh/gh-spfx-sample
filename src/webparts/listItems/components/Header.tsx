import * as React from 'react';

export interface IHeaderProps {}

export interface IHeaderState {}

export default class Header extends React.Component<IHeaderProps, IHeaderState> {
  public render(): React.ReactElement<IHeaderProps> {
    return (
      <div>
        <h1>Welcome to the Demo</h1>
      </div>
    );
  }
}
