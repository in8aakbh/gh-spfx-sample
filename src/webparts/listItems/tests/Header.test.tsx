/// <reference types='jest' />

import * as React from 'react';
import { shallow, configure, ShallowWrapper } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

import Header, { IHeaderProps, IHeaderState } from '../components/Header';

describe('tests for Header component', () => {
  let headerComponent: ShallowWrapper<IHeaderProps, IHeaderState>;

  beforeEach(() => {
    headerComponent = shallow(<Header />);
  });

  afterEach(() => {
    headerComponent.unmount();
  });

  test('should match the existing snapshot', () => {
    expect(toJson(headerComponent)).toMatchSnapshot();
  });
});
