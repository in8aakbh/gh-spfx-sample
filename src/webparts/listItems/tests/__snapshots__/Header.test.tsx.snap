// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Header component should match the existing snapshot 1`] = `
<div>
  <h1>
    Welcome to the Demo
  </h1>
</div>
`;
